manifests/server/config.pp

```diff
@@ -0,0 +1,61 @@
+# This type creates custom configuration files for clickhouse-server.
+#
+# @summary generates xml config from hash via ruby xml-simple
+#
+# @param data
+#   This hash will be converted into xml config placed in `$clickhouse::server::conf_d_dir`.
+#
+#   Root will be `<yandex>` by default.
+# @param section
+#   Should the file be created in configs or users include directory.
+# @param ensure
+#   Subset of attribute `ensure` for `file` type.
+# @param mode
+#   Desired permissions mode for the config file, see `mode` attribute for `file` resource.
+#
+# @example Usage
+#   clickhouse::server::config { 'macros':
+#     data    => {'macros' => [
+#         {'hostname' => ['host.domain.TLD'],}
+#     ]},
+#     section => 'config',
+#   }
+#   #
+#   # Will create file `/etc/clickhouse-server/conf.d/macros.xml`:
+#   # <yandex>
+#   #   <macros>
+#   #     <hostname>host.domain.TLD</hostname>
+#   #   </macros>
+#   # </yandex>
+#
+# @author InnoGames GmbH
+#
+define clickhouse::server::config (
+    Hash                    $data,
+    Enum['config', 'users'] $section,
+    Enum[
+        'present',
+        'file',
+        'absent'
+    ]                       $ensure   = 'present',
+    String[1]               $mode     = '0644',
+) {
+
+    include clickhouse::server
```

Comment:
this doesn't make sense to me. the inheritance should be the other way around, `server` requires `server::config`. a server needs a config to run but a config doesn't need a server. and also from a users' perspective you would define config first and then define the server

Reply:
this also means rearranging files and hierarchies i am afraid

Reply:
as well as for client, everything is correct